lib/matplotlib/font_manager.py

```diff
@@ -1111,21 +1112,12 @@ def score_weight(self, weight1, weight2):
         the CSS numeric values of *weight1* and *weight2*, normalized between
         0.05 and 1.0.
         """
-
         # exact match of the weight names, e.g. weight1 == weight2 == "regular"
-        if (isinstance(weight1, str) and
-                isinstance(weight2, str) and
-                weight1 == weight2):
+        if cbook._str_equal(weight1, weight2):
             return 0.0
-        try:
-            weightval1 = int(weight1)
-        except ValueError:
-            weightval1 = weight_dict.get(weight1, 500)
-        try:
-            weightval2 = int(weight2)
-        except ValueError:
-            weightval2 = weight_dict.get(weight2, 500)
-        return 0.95*(abs(weightval1 - weightval2) / 1000.0) + 0.05
+        w1 = weight1 if isinstance(weight1, Number) else weight_dict[weight1]
+        w2 = weight2 if isinstance(weight2, Number) else weight_dict[weight2]
```

Comment:
They'd get a KeyError on the failing dict lookup, which may not be the nicest but should not be *too* confusing either.  Frankly I'd rather wait for some version of https://github.com/matplotlib/matplotlib/pull/12232 (up to API discussion) to go in rather than add more custom error messages to this kind of things...